### PR TITLE
#128 - IndexProxySlice created

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #128 
I've updated asto, implemented `IndexProxySlice` and extended `PyProxySliceITCase`. `PyProxySlice` now excepts `Storage` instead of `Cache` as different `Cache` implementations will be used in different proxy endpoints.

### CODE REVIEW

File: src/test/java/com/artipie/pypi/http/PyProxySliceITCase.java [hunk 1]
```diff
@@ -68,52 +72,72 @@
...
     @Test
-    void proxiesNotFoundRequest() throws Exception {
+    void returnsErrorIfRecourseNotFound() throws Exception {
         final HttpURLConnection con = (HttpURLConnection) new URL(
             String.format("http://localhost:%s/abc/123/", this.port)
         ).openConnection();
         con.setRequestMethod(RqMethod.GET.value());
         MatcherAssert.assertThat(
-            "Response status is 404",
+            "Response status is 500",
```
Comment (olegmoz) [inline src/test/java/com/artipie/pypi/http/PyProxySliceITCase.java:133]: @olenagerasimova this does not look like proper behavior, I guess proxy should return `404` when artifact cannot be found

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one comment

**Comment by olenagerasimova** [comment]
@olegmoz thanks, agreed and corrected

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, approved

**Comment by 0crat** [comment]
Job `gh:artipie/pypi-adapter#133` is not assigned, can't get performer

**Pull request merged**